src/Orleans/Messaging/IncomingMessageBuffer.cs

```diff
@@ -128,6 +131,19 @@ public bool TryDecodeMessage(out Message msg)
             headerLength = 0;
             bodyLength = 0;
 
+            return true;
```

Comment:
so the `UpdateReceivedData` method is no longer proactively releasing buffers, and instead we release on the next read. Could this lead to leaking buffers if nobody attempts a read afterwards? Would we ALWAYS expect a new read in the future (unless we are fully shutting down Orleans)?